src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private int size = 0;
+    private MyNode<K,V>[] bucketArray = new MyNode[DEFAULT_INITIAL_CAPACITY];
+
+    class MyNode<K,V> {
+        private int hash;
+        private K key;
+        private V value;
+        private MyNode<K, V> next;
+
+        public MyNode(int hash, K key, V value, MyNode<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public MyNode() {
+
+        }
+    }
+
+    public MyHashMap() {
+
+    }
 
     @Override
     public void put(K key, V value) {
+        int arrayIndex = arrayIndexCalculation(key);
+        MyNode<K,V> tempNode = new MyNode<>();
+        MyNode<K,V> newNode = new MyNode<>(key != null ? key.hashCode() : 0, key, value, null);
+        tempNode = bucketArray[arrayIndex];
 
+        do {
+            if (tempNode == null) {
+                bucketArray[arrayIndex] = newNode;
+                size++;
+                break;
+            } else if (newNode.key == null && tempNode.key == newNode.key) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (newNode.key == null && tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            } else if (tempNode.key != null && tempNode.key.equals(newNode.key)) {
+                tempNode.value = newNode.value;
+                break;
+            } else if (tempNode.next == null) {
+                tempNode.next = newNode;
+                size++;
+                break;
+            }
```

Comment:
1st and 3rd IF have the same body - they can be combined into one IF using || operator

After `tempNode.value = newNode.value;` you can use `return`. And then size++; you can do once at the end of the method.